src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -253,23 +253,19 @@ def get_workspace_triad():
 
 
 def list_deployment_connections(
-    subscription_id,
-    resource_group_name,
-    workspace_name,
+    subscription_id="",
+    resource_group_name="",
+    workspace_name="",
     connection="",
 ):
-    try:
-        # Do not support dynamic list for local.
-        from azure.mgmt.cognitiveservices import CognitiveServicesManagementClient
-        from promptflow.azure.operations._arm_connection_operations import \
-            ArmConnectionOperations, OpenURLFailedUserError
-    except ImportError:
-        return None
-
     # For local, subscription_id is None. Does not support dynamic list for local.
-    if not subscription_id:
+    if not subscription_id or (not is_on_runtime()):
```

Comment:
The conditions should be decomposed to two:
1. no workspace triad is available.
2. in local.

Reply:
local: connection is also in local. Despite setting ws triple, conn sub id is not accurate